src/Components/Pagination/Pagination.js

```diff
@@ -0,0 +1,134 @@
+import { useState, useEffect } from 'react';
+import styled from 'styled-components';
+
+const INIT_SETTING = {
+  CURRENT_PAGE: 1,
+  PAGE_SIZE: 5,
+  VIEW_PAGE: 5,
+};
+
+export default function Pagination({
+  items,
+  current = 1,
+  pageSize,
+  handlePage,
+  outputItem,
+}) {
+  const [currentPage, setCurrentPage] = useState(current);
+
+  useEffect(() => {
+    const page = pageSetting(items.length, currentPage, pageSize);
+    const currentPageList = items.slice(page.startIndex, page.endIndex + 1);
+
+    outputItem(currentPageList);
+    handlePage(page);
+  }, [currentPage, items]);
+
+  const pageSetting = (
+    totalItemsRecord,
+    currentPage = INIT_SETTING.CURRENT_PAGE,
+    pageSize = INIT_SETTING.PAGE_SIZE
+  ) => {
+    const totalPages = Math.ceil(totalItemsRecord / pageSize);
+
+    let startPage, endPage;
+
+    if (totalPages <= INIT_SETTING.VIEW_PAGE) {
+      startPage = 1;
+      endPage = totalPages;
+    } else {
+      const centerPage = Math.round(INIT_SETTING.VIEW_PAGE / 2);
+      const rangePage = Math.floor(INIT_SETTING.VIEW_PAGE / 2);
+
+      if (currentPage <= centerPage) {
+        startPage = 1;
+        endPage = INIT_SETTING.VIEW_PAGE;
+      } else if (currentPage + rangePage >= totalPages) {
+        startPage = totalPages - rangePage * 2;
+        endPage = totalPages;
+      } else {
+        startPage = currentPage - rangePage;
+        endPage = currentPage + rangePage;
+      }
+    }
+
+    const startIndex = (currentPage - 1) * pageSize;
+    const endIndex = Math.min(startIndex + pageSize - 1, totalItemsRecord - 1);
+
+    const pages = [...Array(endPage + 1 - startPage).keys()].map(
+      i => startPage + i
+    );
+
+    return {
+      totalItemsRecord: totalItemsRecord,
+      currentPage: currentPage,
+      pageSize: pageSize,
+      totalPages: totalPages,
+      startPage: startPage,
+      endPage: endPage,
+      startIndex: startIndex,
+      endIndex: endIndex,
+      pages: pages,
+    };
+  };
+
+  const pageMove = (moveingPage = 1) => {
+    const totalPages = pageSetting(items.length).totalPages;
+
+    if (moveingPage < 1 || moveingPage > totalPages) {
+      return;
+    }
```

Comment:
```suggestion
    const isPageCanMove = moveingPage >= 1 || moveingPage <= totalPages

    if (!isPageCanMove) return;
```
- 조건을 변수로 할당하고 변수명으로 역할을 설명해두면 어떨까요?
- 조건이 true일 때 움직이지 않는다는게 어색한데 로직을 반대로 적용하면 어떨까요?

Reply:
네 가능하지 않으면 반환되도록 수정하겠습니다.